tests/build-tests/components.test.ts

```diff
@@ -0,0 +1,106 @@
+import { execa } from 'execa';
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import { afterEach, beforeEach, describe, it } from 'vitest';
+
+import { AddonFixtureHelper } from '../fixture-helper.js';
+import { createAddon, createTmp, install } from '../utils.js';
+
+let packageManager = 'pnpm';
+
+describe('components (build)', () => {
+  describe('co-located JS', () => {
+    let cwd = '';
+    let tmpDir = '';
+    let addonOnlyJS: AddonFixtureHelper;
+
+    beforeEach(async () => {
+      tmpDir = await createTmp();
+
+      console.debug(`Debug test repo at ${tmpDir}`);
+
+      let { name } = await createAddon({
+        args: [`--pnpm=true`, '--addon-only'],
+        options: { cwd: tmpDir },
+      });
+
+      cwd = path.join(tmpDir, name);
+
+      addonOnlyJS = new AddonFixtureHelper({
+        cwd,
+        packageManager: 'pnpm',
+        scenario: 'addon-only-js',
+      });
+
+      await install({ cwd, packageManager, skipPrepare: true });
+
+      // NOTE: This isn't needed to make the tests pass atm, but it would be
+      // required when consumers try to use these compiled components
+      await execa('pnpm', ['add', '--save-peer', '@glimmer/component'], { cwd });
+    });
+
+    afterEach(async () => {
+      fs.rm(tmpDir, { recursive: true, force: true });
+    });
+
+    it('generates correct files with no template', async () => {
+      await addonOnlyJS.use('src/components/no-template.js');
+      await addonOnlyJS.build();
+      await addonOnlyJS.matches('dist/components/no-template.js');
```

Comment:
Tbh I am not sure if I would want these kind of snapshot tests against dist-output. Wouldn't these test become super brittle? Like you could update something rollup or babel dependency, which makes the out put slightly different but still perfectly valid, but still you would have tests failing. Especially annoying if you have to fix renovate PRs by hand all the time...

I do see value for fixture-based tests for all the blueprint-generated files, especially those whose content cannot be covered by smoke tests, like e.g. correct path, package manager calls etc. in the generated `CONTRIBUTING.md`.

But for the dist-output we don't have to care what the content really is, we only need to care if it works as expected. And would could cover this by making our smoke tests more detailed, like having different forms of components (TO, colocated, `<template>` and whatnot), and having a test (by means of a fixture?) in the generated addon that would cover this when running the smoke test, right?

Reply:
Yeah, these are good points  -- my thinking with these tests is that they'd be faster than full smoke tests.

But maybe it's still worth it?
Because _ultimately_ we do still want to make sure that each compiled component can be rendered.

I will adjust to more e2e-style